code/game/machinery/kitchen/monkeyrecycler.dm

```diff
@@ -39,79 +39,64 @@
-				return
-			if(target.abiotic())
-				to_chat(user, "<span class='warning'>\the [target] may not have abiotic items on.</span>")
-				return
-			else
-				user.drop_item(G, force_drop = 1)
-				var/ourtype = target.type
-				QDEL_NULL(target)
-				to_chat(user, "<span class='notice'>You stuff \the [target] in the machine.")
-				playsound(src, 'sound/machines/juicer.ogg', 50, 1)
-				use_power(500)
-				src.grinded[ourtype]++
-				to_chat(user, "<span class='notice'>The machine now has [grinded[ourtype]] animals worth of material of this type stored.</span>")
-		else
-			to_chat(user, "<span class='warning'>The machine only accepts animals!</span>")
-	else if(isliving(O))
-		var/mob/living/target = O
+		target = G.affecting
+	/*else if(istype(O, /obj/item/weapon/holder))
+		var/obj/item/weapon/holder/H = O
+		target = H.stored_mob*/
+	if(istype(target))
 		if((target.key || target.ckey) && !emagged)
-			to_chat(user, "<span class='warning'>\the [target] is too sapient for the recycler.</span>")
-			return
-		if(target.stat == CONSCIOUS && !can_recycle_live)
-			to_chat(user, "<span class='warning'>\the [target] is struggling far too much to put it in the recycler.</span>")
-			return
-		if(target.abiotic())
-			to_chat(user, "<span class='warning'>\the [target] may not have abiotic items on.</span>")
-			return
+			failmsg = "\the [target] is too sapient for \the [src]."
+		else if(target.stat == CONSCIOUS && !can_recycle_live)
+			failmsg = "\the [target] is struggling far too much to put it in \the [src]."
+		else if(target.abiotic(1))
+			failmsg = "\the [target] may not have abiotic items on in \the [src]."
 		else
+			if(user) // necessary line, or else the holder or grab could be spammed to abuse this for more monkey cubes
+				user.drop_item(O,force_drop = 1)
 			var/ourtype = target.type
-			QDEL_NULL(target)
-			to_chat(user, "<span class='notice'>You stuff \the [target] in the machine.</span>")
+			if(emagged)
+				for(var/datum/body_archive/archive in body_archives)
+					if(archive && archive.key == target.key)
+						ref_body_archives.Add(archive)
+			qdel(target)
+			if(user)
+				to_chat(user, "<span class='notice'>You stuff \the [target] in the machine.</span>")
 			playsound(src, 'sound/machines/juicer.ogg', 50, 1)
 			use_power(500)
 			src.grinded[ourtype]++
-			to_chat(user, "<span class='notice'>The machine now has [grinded[ourtype]] animals worth of material of this type stored.</span>")
+			if(user)
+				to_chat(user, "<span class='notice'>\the [src] now has [grinded[ourtype]] animals worth of material of this type stored.</span>")
+			else
+				visible_message("<span class='notice'>\the [src] now has [grinded[ourtype]] animals worth of material of this type stored.</span>")
+			return TRUE
+	else
+		failmsg = "\the [src] only accepts animals!"
+	if(failmsg)
+		failmsg = "<span class='warning'>[failmsg]</span>"
+		if(user)
+			to_chat(user, failmsg)
+		else
+			visible_message(failmsg)
```

Comment:
I'm not sure I'm a fan of this, since It can be called on conveyor_act. That means funny spam machines...

Reply:
can remove it if it's that contentious, but your call

Reply:
It can flood visible messages to everyone around on a fully stacked conveyor, so probably not a good idea to have this.

Reply:
done